test/unit/cmap/connection.test.ts

```diff
@@ -137,6 +158,135 @@ describe('new Connection()', function () {
     });
   });
 
+  describe('#onMessage', function () {
+    context('when the connection is a monitoring connection', function () {
+      let queue: Map<number, OperationDescription>;
+      let driverSocket: FakeSocket;
+      let connection: Connection;
+
+      beforeEach(function () {
+        driverSocket = sinon.spy(new FakeSocket());
+      });
+
+      context('when multiple hellos exist on the stream', function () {
+        let callbackSpy;
+        const inputStream = new Readable();
+        const document = { ok: 1 };
+
+        beforeEach(function () {
+          callbackSpy = sinon.spy();
+          const firstHello = generateOpMsgBuffer(document);
```

Comment:
I think we should differentiate the 3 hello docs so that we can check that the correct expected document is being processed, something like `generateOpMsgBuffer({ ...document, id: 1 })`

Reply:
I've updated this so the last hello is now `{ isWritablePrimary: true }` now to distinguish it.